### PR TITLE
Provide guard when ASAN sub-system is missing

### PR DESCRIPTION
**Pull request by mcfadden8**

### CODE REVIEW

**Comment by mcfadden8** [comment]
LGTM

File: src/umpire/CMakeLists.txt [hunk 1]
```diff
@@ -154,6 +154,22 @@ if (ENABLE_OPENMP_TARGET)
   set(UMPIRE_ENABLE_DEVICE On)
 endif ()
 
+include(CheckCXXSourceCompiles)
```
Comment (davidbeckingsale) [inline src/umpire/CMakeLists.txt:157]: Can you please move this to the cmake/ folder.

File: src/umpire/util/memory_sanitizers.hpp [hunk 2]
```diff
@@ -7,6 +7,10 @@
 #ifndef UMPIRE_memory_sanitizers_HPP
 #define UMPIRE_memory_sanitizers_HPP
 
+#include "umpire/config.hpp"
+
+#if defined(UMPIRE_HAS_ASAN)
```
Comment (davidbeckingsale) [inline src/umpire/util/memory_sanitizers.hpp:12]: What's the point of leaving all the extra logic in this file if we already know the header is available?

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]

File: src/umpire/util/memory_sanitizers.hpp (hunk 2 above)
Comment (mcfadden8) [reply src/umpire/util/memory_sanitizers.hpp:12]: Agreed.  Removed

**Comment by mcfadden8** [comment]
LGTM

**Review by davidbeckingsale — APPROVED** [review]

**Comment by mcfadden8** [comment]
LGTM

**Comment by mcfadden8** [comment]
LGTM

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**